src/test/java/org/fogbowcloud/manager/core/models/linkedlist/SynchronizedDoublyLinkedListTest.java

```diff
@@ -0,0 +1,242 @@
+package org.fogbowcloud.manager.core.models.linkedlist;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.fail;
+
+import org.fogbowcloud.manager.core.models.orders.ComputeOrder;
+import org.fogbowcloud.manager.core.models.orders.Order;
+import org.junit.Before;
+import org.junit.Test;
+
+public class SynchronizedDoublyLinkedListTest {
+
+	private SynchronizedDoublyLinkedList list;
+	
+	@Before
+	public void initialize() {
+		this.list = new SynchronizedDoublyLinkedList();
+	}
+	
+	@Test
+	public void testAddFirst() {
+		assertNull(this.list.getHead());
+		assertNull(this.list.getTail());
+		
+		Order order = createOrder("0");
+		this.list.addItem(order);
+
+		assertEquals(order, this.list.getHead().getOrder());
+		assertEquals(order, this.list.getCurrent().getOrder());
+		assertEquals(order, this.list.getTail().getOrder());
+	}
+	
+	@Test
+	public void testAddOrder() {
+		assertNull(this.list.getNext());
+		
+		Order orderOne = createOrder("one");
+		Order orderTwo = createOrder("two");
+		this.list.addItem(orderOne);
+		this.list.addItem(orderTwo);
+		
+		assertEquals(orderOne, this.list.getNext());
```

Comment:
Instead of this.list.getNext() use this.list.resetPointer().

Reply:
Did you mean to use this.list.resetPointer() before the assertEquals? Doesn't make sense to replace list.getNext(), once resetPointer() is a void method.